src/style/style.js

```diff
@@ -311,6 +311,9 @@ class Style extends Evented {
 
         this._loaded = true;
         this.stylesheet = clone(json);
+
+        // store initial projection as prevProjection for checking clearing background tiles in `map.updateProjection`
+        this.map._prevProjection = this.map.getProjection();
```

Comment:
Maybe this could be at the beginning of `_updateMapProjection` so that it's not necessary to add it there?

Reply:
The issue is `style._updateMapProjection` is called in `style.setState` as well, but by the time `style._updateMapProjection` is called in `style.setState`, the result of `map.getProjection` is already updated to the current projection since the stylesheet is updated to the `nextState` in `style.setState`, so it is unable to get the previous map projection with this method at that time. So the same goes for `map.updateProjection` - by the time `setStyle` leads to `map.updateProjection` through `style.updateMapProjection`, it already updated `getProjection` to the current projection.

Reply:
Removed this logic and replaced with a forceSymbolLayerUpdate boolean argument for `updateProjection`